cmd/dra-example-kubeletplugin/main.go

```diff
@@ -141,19 +141,24 @@ func StartPlugin(ctx context.Context, config *Config) error {
 		return fmt.Errorf("path for cdi file generation is not a directory: '%v'", err)
 	}
 
-	driver, err := NewDriver(ctx, config)
-	if err != nil {
-		return err
-	}
-
-	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
-	<-sigc
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	var driver *driver
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		cancel()
+		if err := driver.Shutdown(); err != nil {
+			klog.Errorf("Unable to cleanly shutdown driver: %v", err)
+		}
+	}()
 
-	err = driver.Shutdown(ctx)
```

Comment:
So I looked at the updated description and it is still not clear to me why this change is needed. Why did the old code not work?

As far as I can tell, the change just shuffles code around. Before and after, `driver.Shutdown` should have been reached. What am I missing?